crates/top/rerun/src/commands/rrd/merge_compact.rs

```diff
@@ -226,21 +199,21 @@ fn merge_and_compact(
     )
     .context("couldn't encode messages")?;
 
-    let rrd_out_size = rrd_out.metadata().ok().map(|md| md.len());
+    rrd_out.flush().context("couldn't flush output")?;
 
-    let compaction_ratio =
-        if let (Some(rrds_in_size), Some(rrd_out_size)) = (rrds_in_size, rrd_out_size) {
-            format!(
-                "{:3.3}%",
-                100.0 - rrd_out_size as f64 / (rrds_in_size as f64 + f64::EPSILON) * 100.0
-            )
-        } else {
-            "N/A".to_owned()
-        };
+    let rrds_in_size = rx_size_bytes.recv().ok();
+    let compaction_ratio = if let (Some(rrds_in_size), rrd_out_size) = (rrds_in_size, rrd_out_size)
+    {
+        format!(
+            "{:3.3}%",
+            100.0 - rrd_out_size as f64 / (rrds_in_size as f64 + f64::EPSILON) * 100.0
```

Comment:
This not a ratio, but a reduction metric.

If the input is 100M and the output 25M, the [compression ratio](https://en.wikipedia.org/wiki/Data_compression_ratio) is 4x.

If you want to use the `1 - after/before` metric, you should call it "size reduction" or similar, and add a negation sign to it (100M -> 25M = -75%)

```suggestion
        let size_reduction =
            if let (Some(rrds_in_size), rrd_out_size) = (rrds_in_size, rrd_out_size) {
                format!(
                    "-{:3.3}%",
                    100.0 - rrd_out_size as f64 / (rrds_in_size as f64 + f64::EPSILON) * 100.0
```